workbox-precaching integration test needs refactoring

**Library Affected**:
*workbox-sw, workbox-build, etc.*

I inadvertently left the `workbox-precaching` integration test out of the refactoring work I did in https://github.com/GoogleChrome/workbox/pull/1284, and it's a little flakey as a result.

On further inspection, I am still seeing a decent amount of flakiness in [Travis CI](https://travis-ci.org/GoogleChrome/workbox/builds/340009449#L4272) running one of the precaching tests in Firefox Beta:

```sh
[Workbox Log]: Running Integration test on test/workbox-precaching/integration with NODE_ENV 'dev' and browser 'Firefox Beta'
  [workbox-precaching] Precache and Update
    1) should load a page with service worker 
  0 passing (1s)
  1 failing
  1) [workbox-precaching] Precache and Update
       should load a page with service worker :
      AssertionError: expected 4 to equal 3
      + expected - actual
      -4
      +3
      
      at Context.<anonymous> (test/workbox-precaching/integration/precache-and-update.js:87:103)
      at <anonymous>
      at process._tickCallback (internal/process/next_tick.js:160:7)
```

--- a/test/workbox-precaching/integration/precache-and-update.js
+++ b/test/workbox-precaching/integration/precache-and-update.js
@@ -8,8 +8,6 @@ describe(`[workbox-precaching] Precache and Update`, function() {
   const testingUrl = `${testServerAddress}/test/workbox-precaching/static/precache-and-update/`;
 
   beforeEach(async function() {
-    global.__workbox.server.reset();
-
     // Navigate to our test page and clear all caches before this test runs.
     await cleanSWEnv(global.__workbox.webdriver, testingUrl);
   });
@@ -32,25 +30,24 @@ describe(`[workbox-precaching] Precache and Update`, function() {
     const SW_1_URL = `${testingUrl}sw-1.js`;
     const SW_2_URL = `${testingUrl}sw-2.js`;
 
-    const getIdbData = global.__workbox.seleniumBrowser.getId() === 'safari' ? require('../utils/getPrecachedIDBData-safari') : require('../utils/getPrecachedIDBData');
-
-    // Load the page and wait for the first service worker to register and actsw-testinivate.
-    await global.__workbox.webdriver.get(testingUrl);
+    const getIdbData = global.__workbox.seleniumBrowser.getId() === 'safari' ?
+      require('../utils/getPrecachedIDBData-safari') :
+      require('../utils/getPrecachedIDBData');
 
     // Precaching will cache bust with a search param in some situations.
     const needsCacheBustSearchParam = await global.__workbox.webdriver.executeScript(() => {
       return !('cache' in Request.prototype);
     });
 
+    // Clear out the counters so that we start fresh.
+    global.__workbox.server.reset();
+
     // Register the first service worker.
     await activateSW(SW_1_URL);
 
     // Check that only the precache cache was created.
     const keys = await global.__workbox.webdriver.executeAsyncScript((cb) => {
-      caches.keys()
-      .then((keys) => {
-        cb(keys);
-      });
+      caches.keys().then((keys) => cb(keys));
     });
     expect(keys).to.deep.equal([
       'workbox-precache-http://localhost:3004/test/workbox-precaching/static/precache-and-update/',
@@ -77,14 +74,12 @@ describe(`[workbox-precaching] Precache and Update`, function() {
 
     // Make sure the requested URL's include cache busting search param if needed.
     let requestsMade = global.__workbox.server.getRequests();
-    // This is two due to activateSw() refreshing the page.
-    expect(requestsMade['/test/workbox-precaching/static/precache-and-update/']).to.equal(2);
     if (needsCacheBustSearchParam) {
       expect(requestsMade['/test/workbox-precaching/static/precache-and-update/styles/index.css']).to.equal(1);
       expect(requestsMade['/test/workbox-precaching/static/precache-and-update/index.html?_workbox-cache-bust=1']).to.equal(1);
       expect(requestsMade['/test/workbox-precaching/static/precache-and-update/styles/index.css?_workbox-cache-bust=1']).to.equal(1);
     } else {
-      expect(requestsMade['/test/workbox-precaching/static/precache-and-update/styles/index.css']).to.equal(3);
+      expect(requestsMade['/test/workbox-precaching/static/precache-and-update/styles/index.css']).to.equal(1);
       expect(requestsMade['/test/workbox-precaching/static/precache-and-update/index.html']).to.equal(1);
     }
 

